dbt/adapters/cache.py

```diff
@@ -0,0 +1,463 @@
+                    .format(old_key, new_key, cached.key())
+                )
+                cached.rename_key(old_key, new_key)
+
+        self.relations[new_key] = relation
+
+    def rename(self, old_schema, old_identifier, new_schema,
+               new_identifier):
+        """Rename the old schema/identifier to the new schema/identifier and
+        update references.
+
+        If the new schema/identifier is already present, that is an error.
+        If the schema/identifier key is absent, we only debug log and return,
+        assuming it's a temp table being renamed.
+
+        :param str old_schema: The existing schema name.
+        :param str old_identifier: The existing identifier name.
+        :param str new_schema: The new schema name.
+        :param str new_identifier: The new identifier name.
+        :raises InternalError: If the new key is already present.
+        """
+        old_key = ReferenceKey(
+            schema=old_schema,
+            identifier=old_identifier
+        )
+        new_key = ReferenceKey(
+            schema=new_schema,
+            identifier=new_identifier
+        )
+        logger.debug('Renaming relation {!s} to {!s}'.format(
+            old_key, new_key)
+        )
+        logger.debug('before rename: {}'.format(
+            pprint.pformat(self.dump_graph()))
+        )
+        with self.lock:
+            self._rename_relation(old_key, new_key)
+        logger.debug('after rename: {}'.format(
+            pprint.pformat(self.dump_graph()))
+        )
+
+    def _get_cache_value(self, schema, identifier):
+        """Get the underlying cache value. Raises a KeyError if it does not
+        exist. This is intended for unit testing, mostly.
+
+        :param str schema: The (case-sensitive!) schema name to look for.
+        :param str identifier: The identifier to look for.
+        :return DefaultRelation: The matching relation.
+        :raises KeyError: If the referenced value does not exist.
+        """
+        key = ReferenceKey(schema=schema, identifier=identifier)
+        return self.relations[key]
+
+    def get_relations(self, schema):
+        """Case-insensitively yield all relations matching the given schema.
+
+        :param str schema: The case-insensitive schema name to list from.
+        :return List[DefaultRelation]: The list of relations with the given
+            schema
+        """
+        schema = schema.lower()
+        with self.lock:
+            results = [
+                r.inner for r in self.relations.values()
+                if r.schema.lower() == schema
+            ]
+
+        if None in results:
+            dbt.exceptions.raise_cache_inconsistent(
+                'in get_relations, a None relation was found in the cache!'
+            )
+        return results
+
+    def clear(self):
+        """Clear the cache"""
+        with self.lock:
+            self.relations.clear()
+            self.schemas.clear()
```

Comment:
in general, I think you did a nice job with this module. asks:

- a lot of the comments repeat what is in the docstrings. can you give this a once over and clean some of those up?
- these classes contain public, private, and unit-test-only APIs. as a general rule, I don't like providing functions for unit tests only as I think it complicates the APIs. but the functions here used for unit tests only are marked as such so idc so much. can you just give this a once over and make sure that the public and private APIs are clearly designataed as such?
- it seems like there is a LOT of redundant logging that was probably useful during development, but should perhaps be removed now. can you give these cache logger debug calls a once over?
- finally these APIs work with both (schema,identifier) pairs and relations. i'd prefer to use relations where possible. the best example here that i can see is `rename()` -- we specifically switched `adapter.rename` to use relations instead of (schema,identifier) pairs, so it seems undesirable to me to have the cache class use `old_schema,old_identifier,new_schema,new_identifier` as arguments